### PR TITLE
Huge design overhaul

### PR DESCRIPTION
**Pull request by shroudedcode**
This is my huge design overhaul of the c:geo website, a full list of all my changes is available on the #138 issue. Please leave your comments there unless they are code-specific.

### CODE REVIEW

File: _sass/article.scss [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		&:not(:last-child) {
+			margin-bottom: 16px;
+		}
+	}
+
+    p {
+        line-height: 1.4;
+    }
+
+	ul, ol { margin-left: 35px; }
```
Comment (SammysHP) [inline _sass/article.scss]: Indentation?

File: _sass/article.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    ul { list-style-type: disc; }
+	ol { list-style-type: decimal; }
+    li { margin-bottom: 15px; }
+
+    a {
+		cursor: pointer;
+        padding-bottom: 1px;
```
Comment (SammysHP) [inline _sass/article.scss]: Indentation?

File: _sass/components/actions.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
+.action {
```
Comment (SammysHP) [inline _sass/components/actions.scss]: This files uses tabs instead of spaces for indentation.

File: faq.html [hunk 4]
```diff
@@ -1,93 +1,81 @@
...
+<h1>FAQ</h1>
 
-<div id="language-selection" style="display: none;">
-    <a href="javascript:switchLanguage('en');">English</a>
-    <a href="javascript:switchLanguage('de');">Deutsch</a>
-</div>
+<p>Welcome to our FAQ where we collect and answer frequently asked questions.
+To quickly find an answer to your question, use the <strong>Find in page</strong>
+feature of your browser - usually accessible by pressing <kbd>Ctrl+F</kbd>.</p>
+<p>If you still don't find the answer you're looking for, check out our Support page.</p>
```
Comment (SammysHP) [inline faq.html]: Which is the "Support page" (link it)? Will the user find additional information on that page? Maybe reword a little bit (something like "contact us").

File: faq.html [hunk 5]
```diff
@@ -1,93 +1,81 @@
...
+		        <div class="faq-item-content">
+		            {% for language in languages %}
+		                {% if item[language] == nil %}
+		                    {% assign lang = "en" %}
+		                {% else %}
+		                    {% assign lang = language %}
+		                {% endif %}
+		                <div lang="{{ language }}">
+		                    <div class="question">
+								<div class="icon"><div class="mdi mdi-help"></div></div>
```
Comment (SammysHP) [inline faq.html]: Again indentation.

**Comment by SammysHP** [comment]
Some more changes and questions:

- Can you please tell us the source and license of the background image?
- Including scripts from a CDN allows caching and might have other benefits, but it is a pain for NoScript users. ;) Not sure what other people think about this, probably just "don't care" because restrictive NoScript users are rare.
- While I really like the new index page, I don't like the FAQ at all.
  - The new design is too blocky with too large containers
  - It has no TOC.
  - Also it should not use the abbreviation FAQ alone. Actually I'd prefer the full "Frequently Asked Questions"
  - I like the short introduction, but it should be possible to translate it. Maybe even the main heading ("Häufig gestellte Fragen" in German).
- Underlined links are fine, but the line looks to be too low.
- Old URLs are not redirected. I noticed that while clicking on the status and heatmap links on the development page which still contain ".html".
- The heatmap does not work.
- "Get involved" link is relative, but the URL of pages ends with `/`, so links like `/development/development` are generated.

**Comment by SammysHP** [comment]
A few things I really like:

- responsive, mobile friendly design
- index page with lots of graphics and without big text paragraphs
- the footer

**Comment by SammysHP** [comment]
Some more changes:

- Everything looks a bit colorless
- Missing margin in nested lists (e.g. send2cgeo → Script installation)

**Comment by SammysHP** [comment]
I updated the gems because I had problems compiling native extensions with the old versions. You should rebase on master.

File: faq.html (hunk 4 above)
Comment (shroudedcode) [inline faq.html]: There is "contact us" on the Support page (which I didn't create btw).

**Comment by shroudedcode** [comment]
@SammysHP I don't know how to rebase, I don't want to overwrite or roll back any of my changes.

**Comment by shroudedcode** [comment]
@SammysHP I now fixed most of the issues you addressed. Is there anything I've missed?
The background image by the way is from @unsplash under [Creative Commons Zero](https://unsplash.com/license).

**Comment by rsudev** [comment]
Regarding noscript: I am one of those few that normally browse without any scripts enabled. Of course does not everything work, but the content should be visible.
Not sure if I can check this PR WRT this...

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = tab
```
Comment (SammysHP) [inline .editorconfig]: We want spaces. ;) Please also check all files if they use spaces for indentation.

File: _data/actions.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+- title: Contribute
+  subtitle: Help us build c:geo.
+  description: c:geo is and always will be 100% free and open-source. Join our team and help make it better.
+  buttons:
+  - label: Get involved
+    url: development
```
Comment (SammysHP) [inline _data/actions.yml]: Not absolute path.

File: index.html [hunk 8]
```diff
@@ -1,39 +1,50 @@
 ---
 layout: default
-editlink:
+editLink:
   warning: _includes/warning.md
   short description: _includes/description_short.md
```
Comment (SammysHP) [inline index.html]: Are there still long and short descriptions?

**Comment by SammysHP** [comment]
For rebasing something like

```
$ git remote add upstream ...
$ git fetch upstream
$ git checkout master # just to be sure
$ git rebase upstream/master
$ git push -f origin master
```

For the future try to work in a different branch. ;) 

But I can do the rebasing when I merge everything.

File: .editorconfig (hunk 6 above)
Comment (shroudedcode) [inline .editorconfig]: What? I thought you wanted tabs? They were used everywhere. I only added an exception for `*.yml` because it's best-practice to use spaces there.

Comment (SammysHP) [inline .editorconfig]: Sorry, then there was a misunderstanding. Spaces for everything please.

Comment (shroudedcode) [inline .editorconfig]: Should I do a huge "Replace in project" then replacing all tabs with 2 (or 4?) spaces?

Comment (SammysHP) [inline .editorconfig]: Reformatting would be more helpful (not sure which IDE you use).

> I thought you wanted tabs? They were used everywhere.

Not in the old code, just in your added files.

Comment (shroudedcode) [inline .editorconfig]: That's weird, I was almost sure the indentation was using tabs from the beginning because I normally prefer spaces as well. :hushed: I'll fix the `.editorconfig` and adjust the indentation then.
I'm using @atom as my "IDE" (text editor) by the way.

**Comment by shroudedcode** [comment]
@SammysHP @rsudev I really hate NoScript. :angry: If it would do what the name says - block scripts - I would be fine with it. But why does it block fonts, styles, images, etc. from CDNs?

It sure helps to enforce "do not track", but lots of people use it because they think (or their wanna-be computer experts in the family tell them that) it protects them from viruses and these "evil scripts". And when a website then turns out to look horrible, it's their fault.

**Comment by shroudedcode** [comment]
Oops. I just accidentally closed the PR. :joy_cat: Is anything else missing now?

**Comment by SammysHP** [comment]
Open issues:

- Redirect old `*.html` URLs
- Fix old links that still link to a `*.html` URL (e.g. in *development*)
- Revert to old FAQ design including the TOC (but keep the introduction message)
- Is it possible to remove the flashing, dashed border around links while clicking on them? (Click on a link, keep the button pressed, move the mouse away and release)
- Redesign links. The border is too low IMHO.
- Do you think it is possible to add some more (subtle) colors?

**Comment by shroudedcode** [comment]
@SammysHP How would you go about redirecting the pages? :confused: My only idea would to create [a custom 404 page](https://help.github.com/articles/creating-a-custom-404-page-for-your-github-pages-site/) and let a script on that page do the work (redirecting to the new page). We could also add a hint to simply remove the `.html` for NoScripters.

**Comment by SammysHP** [comment]
https://help.github.com/articles/redirects-on-github-pages/

**Comment by shroudedcode** [comment]
We would have to add that manually to every page, but it's probably the better solution long-term.

File: _layouts/default.html [hunk 9]
```diff
@@ -1,13 +1,13 @@
...
 
-    {% include head.html %}
+    {% include head %}
 
     <body>
         <nav>
 			{% for item in site.data.navigation %}
 				{% assign is_page = false %}
-				{% assign page_url = page.url | remove: '.html' | remove: '/' %}
+				{% assign page_url = page.url | remove: '' | remove: '/' %}
```
Comment (SammysHP) [inline _layouts/default.html]: Superfluous "remove"?

Comment (shroudedcode) [inline _layouts/default.html]: Yeah, now it's useless.

**Comment by shroudedcode** [comment]
Everything's done, to your question about the colors:

I tried to include some colors, but with the dark theme it's very hard to use colors that don't look weird, have too much/not enough contrast, etc. [Flarum.org](flarum.org) uses colors really well, but their page is light-themed.

**Comment by SammysHP** [comment]
Looks great, but I still don't like the blocky design of the FAQ. I prefer the old version.

What do you think @kumy, @rsudev, @Lineflyer?

New:





Old:

**Comment by shroudedcode** [comment]
@SammysHP Would you be okay with removing the category blocks, but keeping the question blocks?

**Comment by kumy** [comment]
Personally I really like the new design. Great work. No real opinion for the FAQ blocks...

@SammysHP May you please share an url to browse it for ourself? (I may have missed it)

**Comment by SammysHP** [comment]
@kumy 
I have it just local, it's described in the README.

**Comment by shroudedcode** [comment]
@kumy Here you go, I created an ngrok tunnel for you: https://d5dfaf17.eu.ngrok.io

**Comment by samueltardieu** [comment]
It looks like the `/status/` page doesn't display statistics anymore. Isn't jquery missing? Also, from the `/development/` page, the link to the heatmap should be `/heatmap/` (right now it points to `/development/heatmap/` which does not exist).

**Comment by shroudedcode** [comment]
@samueltardieu I fixed both of the issues you mentioned.
Are there any more issues that have to be resolved?

**Comment by kumy** [comment]
Faq page:
* would be nice to have some highlight on topic links (:hover)
 ** Ho, it exists, so light, I didn't noticed it.
* https://d5dfaf17.eu.ngrok.io/faq/#coords-format => wrong numbering

**Comment by shroudedcode** [comment]
@kumy I have no clue what causes that issue with the numbering. I guess it's caused by `kramdown`.
Or what do you think @SammysHP?

**Comment by SammysHP** [comment]
This version of the FAQ is better, thank you! Just a few suggestions:

- The contrast between the FAQ text and the background is rather small. It might be better to make the background a bit darker.
- In the TOC and in the FAQ titles the underlining looks inappropriate/intrusive.
- The margin above and below category headings is the same. A bigger top margin might look better.

**Comment by shroudedcode** [comment]
@SammysHP I've made the text a bit lighter, making the background darker doesn't look good.

**Comment by SammysHP** [comment]
The indentation in markdown requires at least three spaces. I suggest to simply revert to the old indentation of 4 spaces.

**Comment by SammysHP** [comment]
- Category titles in the TOC have the normal text color and are darker than the FAQ items.

  

  I suggest to switch both colors.

- The headings in the FAQ are still underlined.

**Comment by shroudedcode** [comment]
@SammysHP The indentation of

**Comment by shroudedcode** [comment]
Arrgh, I always confuse the "Close and comment" button with "Cancel" ...
I switched the colors, and changed the underlining.

**Comment by shroudedcode** [comment]
@SammysHP Should the links in the TOC be un

**Comment by shroudedcode** [comment]
Is anything else missing?

**Comment by SammysHP** [comment]
Just one tiny request: Can you make the edit link like the previous one? This is mostly just for developers, so it shouldn't be that prominent. ;)

**Comment by shroudedcode** [comment]
I decreased the opacity, the padding and font-size.

**Comment by Bananeweizen** [comment]
I'm late into the game, but I also very much like the changes. So please everyone involved go ahead with bringing this live. :)

**Comment by SammysHP** [comment]
If there are no objections or remarks, I'll clean it up a little bit, squash and rebase it and then push it to master. Thanks for you contribution, @shroudedcode!

**Comment by kumy** [comment]
Yes, thanks @shroudedcode

**Comment by shroudedcode** [comment]
@SammysHP @kumy No problem, I'm glad to help with Web Design related stuff in the future! :smile_cat:

**Comment by SammysHP** [comment]
Merged in 4ac293d7756ce9e98772781e2777b59f751a9233, thanks again!
